### PR TITLE
cleanup `extractQueryParams`

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: lib/router/utils.js [hunk 1]
```diff
@@ -25,19 +25,9 @@ export function merge(hash, other) {
...
-  if (
-    len &&
-    len > 0 &&
-    array[len - 1] &&
-    array[len - 1].hasOwnProperty('queryParams')
-  ) {
-    queryParams = array[len - 1].queryParams;
-    head = slice.call(array, 0, len - 1);
-    return [head, queryParams];
+  let possibleQuery = array[array.length - 1];
```
Comment (rwjblue) [inline lib/router/utils.js]: The prior version supported `array` being `undefined`, and this one doesn't.  Are we 100% certain that all callsites pass a valid array?

File: lib/router/utils.js [hunk 2]
```diff
@@ -25,19 +25,9 @@ export function merge(hash, other) {
...
+  if (possibleQuery && possibleQuery.hasOwnProperty('queryParams')) {
```
Comment (rwjblue) [inline lib/router/utils.js]: I'd prefer to change this to `'queryParams' in possibleQuery` (or use `Object.prototype.hasOwnProperty.call(possibleQuery, 'queryParams')`).

File: lib/router/utils.js (hunk 1 above)
Comment (rwjblue) [reply lib/router/utils.js]: Also, isn't this actually `possibleOptions` (since the "query params" object is nested inside of it)?

**Comment by wagenet** [comment]
Closing due to staleness.

**Pull request closed without merging**